apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -1542,61 +1539,43 @@ def _update_dataframe(
         msg = ", ".join(msgs)
         raise ValueError(f"unsupported type updates: {msg}")
 
-    se = tiledb.ArraySchemaEvolution(sdf.context.tiledb_ctx)
-    for drop_key in drop_keys:
-        se.drop_attribute(drop_key)
-
     arrow_table = df_to_arrow(new_data)
     arrow_schema = arrow_table.schema.remove_metadata()
 
+    _pa_to_str_fmt = {
+        pa.string(): "U",
+        pa.binary(): "Z",
+        pa.large_string(): "U",
+        pa.large_binary(): "Z",
+        pa.int8(): "c",
+        pa.uint8(): "C",
+        pa.int16(): "s",
+        pa.uint16(): "S",
+        pa.int32(): "i",
+        pa.uint32(): "I",
+        pa.int64(): "l",
+        pa.uint64(): "L",
+        pa.float32(): "f",
+        pa.float64(): "g",
+        pa.bool_(): "b",
+    }
+
+    add_attrs = dict()
+    add_enmrs = dict()
     for add_key in add_keys:
         # Don't directly use the new dataframe's dtypes. Go through the
         # to-Arrow-schema logic, and back, as this recapitulates the original
         # schema-creation logic.
         atype = arrow_schema.field(add_key).type
-        dtype = tiledb_type_from_arrow_type(atype)
-
-        enum_label: Optional[str] = None
         if pa.types.is_dictionary(arrow_table.schema.field(add_key).type):
-            enum_label = add_key
-            dt = cast(pd.CategoricalDtype, new_data[add_key].dtype)
-            se.add_enumeration(
-                tiledb.Enumeration(
-                    name=add_key, ordered=atype.ordered, values=list(dt.categories)
-                )
-            )
-
-        filters = tiledb_create_options.attr_filters_tiledb(add_key, ["ZstdFilter"])
-
-        # An update can create (or drop) columns, or mutate existing ones.  A
```

Comment:
This was hard-won knowledge, and valuable for knowledge management. Please dont delete it.